playbooks/tensorflow-volcano-kubeflow-integration-test-k8s/kubeflow_tfbenchmarks.yaml

```
@@ -0,0 +1,217 @@
+                if [[ "Failed Succeeded" =~ "$(echo $job_status)" ]]; then
+                    echo "All TFjobs run complate!"
+                    kubectl -n kf-tfbenchmarks get tfjobs
+                    break
+                else
+                    echo "There are TFjobs still in running or pending status..."
+                    kubectl -n kf-tfbenchmarks get tfjobs
+                fi
+                sleep 30
+            done
+            ' || exit_code=$?
+
+        set -x
+        set +e
+        result_path="{{ ansible_user_dir }}/workspace/test_results/kubeflow/{{ item }}"
+        mkdir -p ${result_path}/{pods_logs,jobs_logs}
+
+        if [ ! -d "${result_path}/../jobs_resources" ]; then
+            mkdir -p ${result_path}/../jobs_resources
+            cp -r kf-tfbenchmarks/*  ${result_path}/../jobs_resources/
+        fi
+        if [ ! -f "${result_path}/../results.csv" ]; then
+            echo 'case,job,started_at,complated_at,duration' >> ${result_path}/../results.csv
+        fi
+
+        jobs_names=$(kubectl -n kf-tfbenchmarks get tfjobs -o custom-columns=NAME:.metadata.name --no-headers |grep benchmark)
+        for job_name in ${jobs_names}; do
+            kubectl -n kf-tfbenchmarks describe tfjob ${job_name} > ${result_path}/jobs_logs/${job_name}.log
+            complated_at=`cat ${result_path}/jobs_logs/${job_name}.log |grep 'Completion Time:'|awk '{print $(NF)}'`
+            started_at=`cat ${result_path}/jobs_logs/${job_name}.log |grep 'Start Time:'|awk '{print $(NF)}'`
+            started_at_ts=`date -d ${started_at} +%s`
+            complated_at_ts=`date -d ${complated_at} +%s`
+            duration=''
+            if [ -n "${complated_at_ts}" ]; then
+                duration=`echo $(($complated_at_ts-$started_at_ts)) |awk '{print strftime("%H:%M:%S",$1)}'`
+            fi
+            echo "{{ item }},${job_name},${started_at},${complated_at},${duration}" >> ${result_path}/../results.csv
+        done
+        jobs_pods=$(kubectl -n kf-tfbenchmarks get pods -o custom-columns=NAME:.metadata.name --no-headers |grep benchmark)
+        for job_pod in ${jobs_pods}; do
+            kubectl -n kf-tfbenchmarks logs ${job_pod} > ${result_path}/pods_logs/${job_pod}.log
+        done
+        kubectl -n kf-tfbenchmarks delete -f "{{ ansible_user_dir }}/workspace/kf-tfbenchmarks/{{ item }}"
+        sleep 180
+        kubectl -n kf-tfbenchmarks get tfjobs -o wide
+      args:
+        executable: /bin/bash
+        chdir: '{{ ansible_user_dir }}/workspace'
+      with_items:
+        # We only test the typical case case4, otherwise, it will take very long time.
+        # - case1
+        # - case2
+        # - case3
+        - case4
+
+    - name: kubeflow env cleanup
+      shell: |
+        set -x
+        export KFAPP="kfapp"
+        pushd ${KFAPP}
+        kfctl delete all --delete_storage
+        popd
+        sleep 60
+        kubectl get pods -o wide -A
+        kubectl -n volcano get pods -o wide
```

Why we get `volcano` namespace pods in kubeflow yaml?
```suggestion
        kubectl -n kubeflow get pods -o wide
```

---

hmmm, my mistake, it will return a empty result...